### ISSUE TITLE
bug(rule): explicit ${=...} splitting triggers quoting/unquoted-var

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#150]
## Summary

The current 20-file reference corpus produces one warning from
`quoting/unquoted-var` for the intentional native-Zsh splitting expression in
`z-shell/zsh-eza:functions/.zsh-eza`:

```zsh
builtin emulate -L zsh ${=${options[xtrace]:#off}:+-o xtrace}
```

Fresh evidence on 2026-08-27 used `zsh-lint/next` at
`fa48a58df7462c71d05eaaa8bc60631c48de96a0` and `zsh-eza/main` at
`242216f714032f57e901920081a9305ef353b5f1`.

## Classification

This is a `zsh-lint` false positive, not a consumer quoting defect. The
released Zsh manual defines `${=spec}` as explicitly requesting word splitting.
The existing rule implementation also intends to exclude explicit splitting
flags, but it does not recognize this nested operator form.

Quoting is not a safe remediation: when `xtrace` is off, the quoted form
retains one empty argument and `emulate` reports `unknown argument`; the
unquoted form correctly contributes no argument.

Official reference:
https://zsh.sourceforge.io/Doc/Release/Expansion.html#index-SH_005fWORD_005fSPLIT_002c-toggle

## Scope

- Add focused analyzer coverage for the exact nested `${=...}` form, including
  both non-empty splitting and empty-word elision behavior.
- Make the analyzer recognize explicit native-Zsh splitting through typed AST
  or parser-boundary metadata, without regex or source-text linting.
- Keep ordinary unquoted scalar expansions diagnosed.
- Update the rule's false-positive documentation if necessary.
- Do not change `zsh-eza`, add a suppression, waive the warning, or downgrade
  its severity.

## Acceptance criteria

- [ ] The minimized explicit-splitting fixture produces no
      `quoting/unquoted-var` warning.
- [ ] Existing bad and good rule cases continue to pass.
- [ ] Native-Zsh controls demonstrate the argument-vector difference between
      the intentional unquoted form and an unsafe quoted rewrite.
- [ ] `go test ./...`, `go vet ./...`, and `go build ./...` pass.
- [ ] The documented 20-file corpus parses 20/20 and reports zero errors and
      zero warnings without suppressions.

Parent roadmap milestone: #132.

**Comment by ss-o** [comment]
Completed by PR #152, squash-merged to next as 6aa9c88d37b3. The explicit native Zsh word-split form no longer triggers quoting/unquoted-var, its inverse form remains covered, and all 30 published checks pass.
